### PR TITLE
FED-2239 Analyze analyzer_plugin in CI, fix warnings

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
I kept seeing analysis warnings locally in the analyzer plugin directory, and figured they must be only happening locally, otherwise they'd be failing CI.

Turns out, we weren't analyzing the analyzer plugin in CI 😅.

## Changes
- Add CI step to run analysis on the analyzer_plugin directory
- Fix all warnings

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        - CI runs analysis and fails on warnings before they were fixed in https://github.com/Workiva/over_react/actions/runs/7908062168/job/21586466107
        - CI passes on latest commit
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart [hunk 1]
```diff
@@ -926,7 +926,7 @@ class ExhaustiveDeps extends DiagnosticContributor {
         }
 
         Expression? maybeID = declaredDependencyNode;
-        while (maybeID is PropertyAccess || maybeID is PrefixedIdentifier) {
+        while (maybeID != null && (maybeID is PropertyAccess || maybeID is PrefixedIdentifier)) {
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart:929]: This `!= null` is redundant, but without it `maybeID` isn't promoted to be non-nullable

File: tools/analyzer_plugin/lib/src/diagnostic/visitors/non_static_reference_visitor.dart [hunk 2]
```diff
@@ -59,6 +59,7 @@ class ReferenceVisitor extends RecursiveAstVisitor<void> {
 bool referencesImplicitThis(SimpleIdentifier identifier) {
   // prepare element
   final element = identifier.staticElement;
+  if (element == null) return false;
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/lib/src/diagnostic/visitors/non_static_reference_visitor.dart:62]: Same here; this null-check is redundant, but helps with promotion.

**Review by sydneyjodon-wk — APPROVED** [review]
+10 oops lol

(pending CI passes)
- [x] CI fails on commit with analysis errors

**Pull request merged**